### PR TITLE
Playwright: Avoid gutenframe in atomic tests

### PR DESCRIPTION
**Pull request by dpasque**
#### Changes proposed in this Pull Request

*

#### Testing instructions

*

Related to #

### CODE REVIEW

**Comment by dpasque** [comment]
Just getting what I have up for now. Going to revisit this with intercepting URLs with `page.route` later.

File: test/e2e/specs/editor/editor__post-advanced-flow.ts [hunk 1]
```diff
@@ -121,7 +129,7 @@ describe( DataHelper.createSuiteTitle( `Editor: Advanced Post Flow` ), function
 		it( 'Post is no longer visible', async function () {
 			const testPage = await browser.newPage();
 			await testPage.goto( postURL.href );
-			await testPage.waitForSelector( 'div.error-404.not-found' );
+			await testPage.waitForSelector( 'body.error404' );
```
Comment (worldomonation) [inline test/e2e/specs/editor/editor__post-advanced-flow.ts:132]: @fullofcaffeine did you find out the answer to this? 
p1648692332838129/1648603003.564999-slack-C01QG4Y91RR

We just need the 404 to work for most cases if the selector does indeed change between themes.
If the difference is due to AT and Simple, we can alternatively do the following:

```
await Promise.race([ 
    testPage.waitForSelector( <simple_selector> ),
    testPage.waitForSelector( <atomic_selector> )
])
```

File: packages/calypso-e2e/src/lib/pages/posts-page.ts [hunk 2]
```diff
@@ -101,8 +106,21 @@ export class PostsPage {
...
 	 */
 	async newPost(): Promise< void > {
-		const locator = this.page.locator( selectors.addNewPostButton );
-		await locator.click();
+		const newPostLocator = this.page.locator( selectors.addNewPostButton );
+
+		// Get ready for some temporary yuckiness!
+		// On atomic sites, the Gutenframe does not play well with our E2E tests.
+		// For the time being, we need to make redirect any click navigations from this Calypso page
+		// to the wp-admin version of the editor.
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/posts-page.ts:114]: I like that it's succinct and this is likely the least yucky we can make it. Nice work.

File: packages/calypso-e2e/src/data-helper.ts [hunk 3]
```diff
@@ -289,3 +289,29 @@ export function createSuiteTitle( title: string ): string {
...
+ * @returns Host name without scheme (e.g. 'testsite.wordpress.com').
+ * @throws If no host name can be parsed.
+ */
+export function parseSiteHostFromUrl( url: string ): string {
+	if ( url.includes( '/wp-admin' ) ) {
+		return new URL( url ).host;
+	}
+
+	const pathSuffix = new URL( url ).pathname.split( '/' ).pop();
+	const hostRegex = /([a-zA-Z0-9-]+\.)+[a-zA-Z0-9-]+/;
```
Comment (worldomonation) [inline packages/calypso-e2e/src/data-helper.ts:309]: 😵

File: test/e2e/specs/editor/editor__post-advanced-flow.ts (hunk 1 above)
Comment (fullofcaffeine) [reply test/e2e/specs/editor/editor__post-advanced-flow.ts:132]: @worldomonation I was able to verify that `body.error404` was present in both my simple and AT test sites 👍🏻 I think we could leave this selector for now unless we find out it starts failing in other test sites.

File: packages/calypso-e2e/src/lib/pages/pages-page.ts [hunk 4]
```diff
@@ -26,15 +27,32 @@ export class PagesPage {
 	 * Example {@link https://wordpress.com/pages}
 	 */
 	async visit(): Promise< Response | null > {
-		const response = await this.page.goto( getCalypsoURL( 'pages' ) );
+		const [ , response ] = await Promise.all( [
+			// We need to wait until the site-specific redirect happens.
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/lib/pages/pages-page.ts:31]: It's not clear why this is needed though. Would you mind elaborating here?

File: packages/calypso-e2e/src/lib/pages/pages-page.ts [hunk 5]
```diff
@@ -26,15 +27,32 @@ export class PagesPage {
...
+			this.page.waitForNavigation( { url: '**/pages/**' } ),
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/lib/pages/pages-page.ts:32]: Is this required for this changeset or is it an unrelated change?

File: packages/calypso-e2e/src/data-helper.ts [hunk 6]
```diff
@@ -289,3 +289,29 @@ export function createSuiteTitle( title: string ): string {
 
 	return parts.join( ' ' );
 }
+
+/**
+ * Parses a WPCOM site host name (without the scheme) from a given site-specific page URL.
+ * For example:
+ *  - testsite.wordpress.com/wp-admin/post --> testsite.wordpress.com
```
Comment (fullofcaffeine) [inline packages/calypso-e2e/src/data-helper.ts:296]: I think it's good that we're handling both URL patterns here! I'm curious though: considering this is now only called from the post page POM - which should be running "from" Calypso - did you actually already stumble upon a scenario where it was called while the top-level site site had a `wp-admin` URL?

**Comment by dpasque** [comment]
We don't need this anymore. Closing this PR
